jobs/release.Jenkinsfile

```diff
@@ -207,6 +207,7 @@ lock(resource: "release-${params.STREAM}", extra: locks) {
         // OCP ART doesn't actually care what the tag name is (it's just to stop GC), we
         // hardcode it.
         def push_containers = ['oscontainer': ['ostree', 'base-oscontainer', ''],
+                               'kubevirt': ['kubevirt', 'kubevirt', ''],
```

Comment:
I think this one is tricky because I think upstream we don't want a tag suffix, but downstream we do.

@jlebon does that match your understanding?

Reply:
Indeed. We can't not add a suffix for this downstream. It'll conflict with the base-oscontainer. Upstream, we could push as `quay.io/fedora/fedora-coreos-kubevirt:stable-kubevirt`, but that looks awkward. Probably time to make this configurable. One idea is to make it templated, like we do for other keys. E.g.:

```yaml
registry_repos:
    kubevirt: quay.io/example/monorepo:${STREAM}
```

or

```yaml
registry_repos:
    kubevirt: quay.io/example/monorepo:${STREAM}-suffix
```

To make it backwards-compatible, we could have the third element in the tuple here represent the default tag template when none is given, so e.g.

```groovy
'extensions': ['extensions-container', 'extensions-container', '${STREAM}-extensions']
```

Reply:
I liked this idea when I first read it, but I'm starting to think it's not complete enough.

we also have the `registry_repos.add_build_tag` piece that adds `--tag=${params.VERSION}${tag_suffix}` so we'd have to split out `tag_suffix` from `quay.io/example/monorepo:${STREAM}-suffix`

Maybe instead we should just add more flexibility in the config spec:

```
registry_repos:
    kubevirt:
        registry: quay.io/example/monorepo
        tags:
          - ${STREAM}-suffix
          - ${VERSION}-suffix
```

IIUC we don't need to preserve backwards compat here because we use the latest pipeline code and latest pipecfg (i.e. there are only one copy of them for all branches and we can update them at the same time).

Reply:
Yup, making the templating bits a list works for me too! The `add_build_tag` knob is definitely a bit of a hack, so it'd be good to get rid of it. The part about backcompat was more to provide a path to make this move more seamless, not that we can't change everything if we wanted to.

Reply:
ok pushed a new upload that should take care of this.